sos/plugins/networking.py

```diff
@@ -17,10 +18,18 @@ class Networking(Plugin):
     plugin_name = "networking"
     profiles = ('network', 'hardware', 'system')
     trace_host = "www.example.com"
-    option_list = [(
+    option_list = [
         ("traceroute", "collect a traceroute to %s" % trace_host, "slow",
-         False)
-    )]
+         False),
+        ("namespaces", "specific namespaces regex to be " +
+         "collected, namespaces should be separated by space " +
+         "as for example \"eth* ens2\"", "fast", ""),
```

Comment:
I think it'd be more user-consumable if the regexes were comma-delimited rather than space-delimited. Using commas is more consistent with how sos handles lists in options already.

Reply:
Comma separated list does not work. Since in https://github.com/sosreport/sos/blob/master/sos/sosreport.py#L175-L178, the `ArgumentParser` splits the string e.g. `networking.namespaces='a*,b*,c*'` to three options `networking.namespaces=a*`, `b*` and `c*'` - it follws the same logic like `-o kernel,boot` that appends to `onlyplugins` two options.

Checking if `ArgumentParser` has a trick for this..

Reply:
It probably has:

diff --git a/sos/sosreport.py b/sos/sosreport.py
index 2d2d1599..4a768f45 100644
--- a/sos/sosreport.py
+++ b/sos/sosreport.py
@@ -172,7 +172,7 @@ def _get_parser():
     parser.add_argument("-e", "--enable-plugins", action="extend",
                         dest="enableplugins", type=str,
                         help="enable these plugins", default=[])
-    parser.add_argument("-k", "--plugin-option", action="extend",
+    parser.add_argument("-k", "--plugin-option", action="append",
                         dest="plugopts", type=str,
                         help="plugin options in plugname.option=value "
                              "format (see -l)", default=[])

This stores whole `-k any_string_with_commas` as one item of the `plugopts` list, which allows `-k networking.namespaces=a*,b*,c*` to set the option value to `a*,b*,c*`.

HOWEVER, the side effect is each and every plugin option must have to be specificed via `-k` option. As `-k networking.traceroute=on,qpid.port=12345` now splits the options to two, while the above change does not.

Reply:
Ah, ok - I was thinking/hoping that ArgParse would be smart enough to take the quoted regex string as a whole unit, but I guess not.

Spaces it is then.